openverse_catalog/docs/adding_a_new_provider.md

```diff
@@ -0,0 +1,90 @@
+
+We provide a [script](../dags/templates/create_provider_ingester.py) that can be used to generate the files you'll need and get you started:
+
+```
+# PROVIDER: The name of the provider
+# ENDPOINT: The API endpoint from which to fetch data
+# MEDIA: Optionally, a space-delineated list of media types ingested by this provider
+#        (and supported by Openverse). If not provided, defaults to "image".
+
+> create_provider_data_ingester.py <PROVIDER> <ENDPOINT> -m <MEDIA>
+
+# Example usage:
+
+> python3 openverse_catalog/templates/create_provider_ingester.py "Foobar Museum" "https://foobar.museum.org/api/v1" -m image audio
+```
+
+You should see output similar to this:
+```
+Creating files in /Users/staci/projects/openverse-projects/openverse-catalog
+API script:        openverse-catalog/openverse_catalog/dags/providers/provider_api_scripts/foobar_museum.py
+API script test:   openverse-catalog/tests/dags/providers/provider_api_scripts/test_foobar_museum.py
+
+NOTE: You will also need to add a new ProviderWorkflow dataclass configuration to the PROVIDER_WORKFLOWS list in `openverse-catalog/dags/providers/provider_workflows.py`.
+```
+
+This generates a provider script with a templated `ProviderDataIngester` for you in the [`provider_api_scripts` folder](../dags/providers/provider_api_scripts), as well as a corresponding test file. Complete the TODOs detailed in the generated files to implement behavior specific to your API.
+
+Some APIs may not fit perfectly into the established `ProviderDataIngester` pattern. For advanced use cases and examples of how to modify the ingestion flow, see the [`ProviderDataIngester` FAQ](provider_data_ingester_faq.md).
+
+
+### Add a `ProviderWorkflow` configuration class
+
+Now that you have an ingester class, you're ready to wire up a provider DAG in Airflow to automatically pull data and load it into our Catalog database. This is as simple as defining a `ProviderWorkflow` configuration dataclass and adding it to the `PROVIDER_WORKFLOWS` list in [`provider_workflows.py`](../dags/providers/provider_workflows.py). Our DAG factories will pick up the configuration and generate a complete new DAG in Airflow!
+
+At minimum, you'll need to provide the following in your configuration:
+* `provider_script`: the name of the file where you defined your `ProviderDataIngester` class
+* `ingestion_callable`: the `ProviderDataIngester` class itself
+* `media_types`: the media types your provider handles
```

Comment:
I'm realizing that we should make an issue for removing the legacy wrapper handling logic, which also means that we should be able to remove `provider_script` here eventually. I can make that issue 🙂 Also, I think we could even determine `media_types` automatically by using `ProviderDataIngester::providers.keys()` 😮 Lots of pairing down we could do on these configs once the refactors are all done! 🥳